scipy/special/tests/test_logsumexp.py

```diff
@@ -196,11 +197,38 @@ def test_dtypes_ab(self, dtype_a, dtype_b, xp):
         xp_assert_close(logsumexp(a, b=b), desired)
 
     def test_gh18295(self, xp):
+        # gh-18295 noted loss of precision when real part of one element is much
+        # larger than the rest. Check that this is resolved.
         a = xp.asarray([0.0, -40.0])
         res = logsumexp(a)
         ref = xp.logaddexp(a[0], a[1])
         xp_assert_close(res, ref)
 
+    @pytest.mark.parametrize('dtype', ['complex64', 'complex128'])
+    def test_gh21610(self, xp, dtype):
+        # gh-21610 noted that `logsumexp` could return imaginary components
+        # outside the range (-pi, pi]. Check that this is resolved.
+        # While working on this, I noticed that all other tests passed even
+        # when the imaginary component of the result was zero. This suggested
+        # the need of a stronger test with imaginary dtype.
```

Comment:
Would be good to include a test with `b`, or maybe just a separate property based test of the basic behavior for all input types and options.